### PR TITLE
Update UIKit with local cache

### PR DESCRIPTION
**Pull request by Steven-Chan**
connect #52

### CODE REVIEW

**Comment by Steven-Chan** [comment]
> I am a bit confuse on the reason on having this.async at RealmStore. And it has no effect on delete related function... will it make the usage on get/delete asymmetry? It seems not a good API to use.

If everything is sync, the android ui would throw a lot of warnings, becoz the load more messages call is mostly likely triggered by a scroll event.

However, the store is also used by test, and I want to keep the test case simple and synchronous, so the realm store has this option.

Oh, I was just being lazy, becoz the warnings only throw on get, I will also make write and delete take care of the `async` flag.

File: chat/src/main/java/io/skygear/plugins/chat/ui/SortedMessageListAdapter.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+            Wrapper next = each.next();
+            if (next.item instanceof Date) {
+                each.remove();
+            }
+        }
+    }
+
+    private void regenerateDateHeaders() {
+        // Assume all date headers are removed
+        List<Integer> indeicesToInsert = new ArrayList<>();
```
Comment (cheungpat) [inline chat/src/main/java/io/skygear/plugins/chat/ui/SortedMessageListAdapter.java]: s/indeices/indices/

**Comment by cheungpat** [comment]
@rickmak I am going to merge this when steven made some minor changes I requested

**Comment by cheungpat** [comment]
I am going to make the changes and merge to master.

**Comment by cheungpat** [comment]
@Steven-Chan sorry the pull request is failing

**Comment by Steven-Chan** [comment]
@cheungpat Updated and travis passed.

**Pull request merged**